File: bank-service/src/main/java/rs/edu/raf/dto/CreateCardDto.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

import java.math.BigDecimal;

public record CreateCardDto (
    @NotNull String type,
    @NotNull Long issuerId,
    @NotNull String name,
    @NotNull Long bankAccountNumber,
    @NotNull @Positive BigDecimal limit

) { }

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CreateCardDTO.kt
package org.example.bankservicekotlin.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import java.math.BigDecimal

data class CreateCardDTO(
    @NotNull val type: String,
    @NotNull val issuerId: Long,
    @NotNull val name: String,
    @NotNull val bankAccountNumber: Long,
    @NotNull @Positive val limit: BigDecimal
)
